crypto/dh/dh_kdf.c

```diff
@@ -185,3 +190,4 @@ int DH_KDF_X9_42(unsigned char *out, size_t outlen,
     EVP_MD_CTX_cleanup(&mctx);
     return rv;
 }
+#endif
```

Comment:
'clang -Weverything' is great at finding this sort of thing.
Note that if the goal is to ensure that one can reproduce the issue being raised here, then -DPEDANTIC should *not* be used, as defining that macro is telling openssl that we are in a pedantic environment and should use the "pedantically correct" code paths, as opposed to the normal ones.  That is, the build should error at every other place where the "dummy" declaration is used, when -DPEDANTIC is not used, and the appropriate compiler flags are given.  Then the same compiler flags can be used against the new code in question to verify that the dummy declaration is needed there.